pkg/notify/notify.go

```diff
@@ -3,166 +3,218 @@ package notify
 import (
 	"bytes"
 	"fmt"
-	"github.com/go-kit/kit/log"
-	"github.com/go-kit/kit/log/level"
 	"io/ioutil"
 	"reflect"
 	"strings"
 	"time"
 
-	"github.com/free/jiralert/pkg/config"
-	"github.com/free/jiralert/pkg/template"
-
 	"github.com/andygrunwald/go-jira"
 	"github.com/free/jiralert/pkg/alertmanager"
+	"github.com/free/jiralert/pkg/config"
+	"github.com/free/jiralert/pkg/template"
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/pkg/errors"
 	"github.com/trivago/tgo/tcontainer"
 )
 
-// Receiver wraps a JIRA client corresponding to a specific Alertmanager receiver, with its configuration and templates.
+// TODO(bwplotka): Consider renaming this package to ticketer.
+
+type jiraIssueService interface {
```

Comment:
Well testing is one thing, but mainly to remove jira dependency from this package. TBH it should not be jira specific, but any ticketing system in theory. It just helps to make things more cleaner and allow better composability.

It's private because I don't expect anyone to use this interface (use = accept it in some other method/struct) anywhere else. There is no harm in leaving it private IMO.

Reply:
It's not popular pattern but I cannot see any issue with this (:

Reply:
> I cannot see any issue with this

Let me put it another way then (it's only a joke, no worries): I will approve this PR when it meets all my criteria. I'm not going to tell you what those are, you should not be burdened with them. (o:

More seriously, like I said, since the compiler seems happy about it I'm OK with it as is.